shunting-yard-replicator/src/main/java/com/hotels/shunting/yard/replicator/exec/event/aggregation/DefaultMetaStoreEventCompactor.java

```diff
@@ -0,0 +1,118 @@
+/**
+ * Copyright (C) 2016-2018 Expedia Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hotels.shunting.yard.replicator.exec.event.aggregation;
+
+import java.util.LinkedList;
+import java.util.List;
+import java.util.ListIterator;
+import java.util.function.Predicate;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import com.hotels.shunting.yard.replicator.exec.event.MetaStoreEvent;
+
+class DefaultMetaStoreEventCompactor {
+  private static final Logger log = LoggerFactory.getLogger(DefaultMetaStoreEventCompactor.class);
+
+  private final EventMerger eventMerger;
+
+  DefaultMetaStoreEventCompactor() {
+    this(new EventMerger());
+  }
+
+  DefaultMetaStoreEventCompactor(EventMerger eventMerger) {
+    this.eventMerger = eventMerger;
+  }
+
+  /**
+   * Reduces the number of events to the minimum necessary to execute Circus Train as few times as possible.
+   * <p>
+   * This method assumes all the events passed in are for the same table and are sorted in chronological order, i.e.
+   * they come from the same source table.
+   * <p>
```

Comment:
I'm going to let the </p> slide for now ;)

Reply:
thanks Adrian